java/src/main/native/src/RmmJni.cpp

```diff
@@ -14,14 +14,118 @@
  * limitations under the License.
  */
 
+#include <rmm/mr/device/default_memory_resource.hpp>
+
 #include "jni_utils.hpp"
 
+namespace {
+
+class java_event_handler_memory_resource : public rmm::mr::device_memory_resource {
```

Comment:
```suggestion
class java_event_handler_memory_resource final : public rmm::mr::device_memory_resource {
```
If you don't intend on further deriving from this class, it's good to use `final`. There's more chance the compiler will be able to devirtualize the calls to `allocate/deallocate`.